src/lib/components/ProjectsPanel/components/Modal.tsx

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useEffect, useState } from "react";
+import classNames from "classnames";
+import FocusLock from "react-focus-lock";
+import styles from "./Modal.module.css";
+import { Styles, Theme } from "../types";
+
+export function useModal() {
+  const [isOpen, setisOpen] = useState(false);
+  const close = useCallback(() => setisOpen(false), []);
+  const toggle = useCallback(() => setisOpen(!isOpen), [setisOpen, isOpen]);
+
+  useEffect(() => {
```

Comment:
**suggestion** Maybe separate in `useEsc()` hook ? (maybe necessary in other components)

Reply:
added lib)